### DISCUSSION TITLE
How often is `train_dataloader()` called for a `LightningDataModule`?

### DISCUSSION THREAD

**Discussion by EricWiener** [Lightning-AI/pytorch-lightning#10383] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi,

I'm seeing my model overfit and I'm trying to figure out what's going wrong. I believe that the model isn't being trained on the entire train dataset. I define my `LightningDataModule` as follows:
```
class MyDataModule(pl.LightningDataModule):
    def train_dataloader(self):
        dataset = MyDataset()
        return DataLoader(MyDataset, num_workers=4)

trainer = Trainer(...)
model = Model(...)
data_module = MyDataModule(...)
trainer.fit(model, data_module)
```
Is it okay that I am re-initiating my dataset every time `train_dataloader` is called? I printed out when `train_dataloader` is called and it seems to only be called once and not get reset when switching from train -> val. Could I confirm this is the actual behavior and there isn't any possibility that `MyDataset()` will end up being re-initialized/reset somehow for training data?

Thank you!

**Comment by rohitgr7** [comment]
yes it will be called only once unless you set `Trainer(reload_datalaoders_every_n_epochs=int > 0)`.
